include/ppx/geometry.h

```diff
@@ -146,8 +153,14 @@ class Geometry
     {
     public:
         Buffer() {}
-        Buffer(BufferType type, uint32_t elementSize)
-            : mType(type), mElementSize(elementSize) {}
+        Buffer(BufferType type, uint32_t elementSize, uint32_t knownElementCount = 0)
+            : mType(type), mElementSize(elementSize), mKnownElementCount(knownElementCount)
+        {
+            if (mKnownElementCount > 0) {
+                // Pre-determined final size of buffer
+                mData.resize(mKnownElementCount * mElementSize);
+            }
```

Comment:
Even though the check adds clarity, `mData.resize()` will just return if you give a 0.

Reply:
Noted, thanks. Changed